libs/features/mas/src/hydrate.js

```diff
@@ -175,17 +296,31 @@ function createSpectrumSwcButton(cta, aemFragmentMapping, isOutline, variant) {
             tabIndex: 0,
             size: aemFragmentMapping.ctas.size ?? 'm',
         },
-        cta,
+        cta.innerHTML,
     );
 
-    spectrumCta.addEventListener('click', (e) => {
-        if (e.target !== cta) {
-            /* c8 ignore next 3 */
+    (async () => {
+        try {
+          const CheckoutButtonEl = customElements.get('checkout-button');
+          const checkoutBtn = CheckoutButtonEl?.createCheckoutButton({}, cta.innerHTML);
+          for (const attr of cta.attributes) {
+              try {
+                  checkoutBtn.setAttribute(attr.name, attr.value);
+              } catch (e) {
+                  console.warn(`Failed to copy attribute ${attr.name}`, e);
+              }
+          }
+          checkoutBtn.connectedCallback();
+          await checkoutBtn.render();
+          await checkoutBtn.onceSettled();
+          spectrumCta.addEventListener('click', (e) => {
             e.stopPropagation();
-            cta.click();
+            checkoutBtn.click(e);
```

Comment:
`click` method does not take parameters.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click